pallets/avn/src/tests/session_handler_tests.rs

```diff
@@ -1,167 +1,272 @@
-                    Validator { account_id: 3, key: UintAuthorityId(3) }
-                ]
-        );
-
-        change_validators_good();
-
-        assert!(
-            AVN::validators() ==
-                vec![
-                    Validator { account_id: 1, key: UintAuthorityId(1) },
-                    Validator { account_id: 2, key: UintAuthorityId(2) }
-                ]
-        );
-    });
-}
+        ext.execute_with(|| {
+            assert!(
+                AVN::validators() ==
+                vec![ TestAccount::derive_validator(1), TestAccount::derive_validator(2), TestAccount::derive_validator(3)]
+            );
+        });
+    }
 
-#[test]
-// * changed is true: Ensure that the keys have been updated
-fn keys_populated_correctly_new_session_with_empty_change() {
-    let mut ext = ExtBuilder::build_default().with_validators().as_externality();
-    ext.execute_with(|| {
-        assert!(
-            AVN::validators() ==
-                vec![
-                    Validator { account_id: 1, key: UintAuthorityId(1) },
-                    Validator { account_id: 2, key: UintAuthorityId(2) },
-                    Validator { account_id: 3, key: UintAuthorityId(3) }
-                ]
-        );
-
-        change_validators_empty();
-
-        assert!(AVN::validators() == vec![]);
-    });
-}
+    #[test]
+    fn if_no_changes_between_sessions_then_avn_knows_same_collators()
+    {
+        let mut ext = setup_initial_collators();
 
-#[test]
-// * changed is false: keys list has not changed
-fn keys_populated_correctly_new_session_with_no_change() {
-    let mut ext = ExtBuilder::build_default().with_validators().as_externality();
-    ext.execute_with(|| {
-        assert!(
-            AVN::validators() ==
-                vec![
-                    Validator { account_id: 1, key: UintAuthorityId(1) },
-                    Validator { account_id: 2, key: UintAuthorityId(2) },
-                    Validator { account_id: 3, key: UintAuthorityId(3) }
-                ]
-        );
-
-        advance_session_no_validators_change();
-
-        assert!(
-            AVN::validators() ==
-                vec![
-                    Validator { account_id: 1, key: UintAuthorityId(1) },
-                    Validator { account_id: 2, key: UintAuthorityId(2) },
-                    Validator { account_id: 3, key: UintAuthorityId(3) }
-                ]
-        );
-    });
-}
+        ext.execute_with(|| {
+            let initial_collators = avn_known_collators();
+
+            advance_session();
+
+            let current_collators = avn_known_collators();
+
+            advance_session();
+
+            assert_eq!(initial_collators, current_collators);
+            assert_eq!(current_collators, vec![
+                TestAccount::derive_validator(1),
+                TestAccount::derive_validator(2),
+                TestAccount::derive_validator(3),
+            ]);
```

Comment:
probably not necessary. We've tested for equality after the session rotation, and we must have checked these contents in another test.